.github/workflows/ci.yaml

```diff
@@ -0,0 +1,77 @@
+name: CI
+on:
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'docs/**'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  unittest:
+    name: unit tests
+    strategy:
+      matrix:
+        python-version: ["3.11","3.12","3.13"]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install ".[test]"
+      - name: Test with pytest
+        # when you change the study base folder, update the below
+        run: |
+          python -m pytest --cov-report xml --cov=cumulus_library_template tests
+      - name: Log missing coverage
+        run: |
+          coverage report -m --skip-covered
+      - name: Generate coverage report
+        if: github.ref != 'refs/heads/main'
+        uses: orgoro/coverage@v3.2
+        with:
+            coverageFile: coverage.xml
+            token: ${{ secrets.GITHUB_TOKEN }}
+            thresholdAll: 0
+            thresholdNew: 0
+            thresholdModified: 0
+
+
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11","3.12","3.13"]
```

Comment:
Why run linters for each python? The results shouldn't change for these linters.